scirpy/datasets/__init__.py

```diff
@@ -255,16 +255,20 @@ def iedb(cached: bool = True, *, cache_path="data/iedb.h5ad") -> AnnData:
         ]
     )
 
-    iedb_df.loc[
-        iedb_df["Chain 1 CDR3 Curated"].isna(), "Chain 1 CDR3 Curated"
-    ] = iedb_df["Chain 1 CDR3 Calculated"]
-
-    iedb_df.loc[
-        iedb_df["Chain 2 CDR3 Curated"].isna(), "Chain 2 CDR3 Curated"
-    ] = iedb_df["Chain 2 CDR3 Calculated"]
-
-    iedb_df["Chain 1 CDR3 Curated"] = iedb_df["Chain 1 CDR3 Curated"].str.upper()
-    iedb_df["Chain 2 CDR3 Curated"] = iedb_df["Chain 2 CDR3 Curated"].str.upper()
+    # If no curated CDR3 sequence or V/D/J gene is available, the calculated one is used.
+    def replace_curated(input_1, input_2):
+        Calculated_1 = input_1.replace("Curated", "Calculated")
+        tcr_table.loc[tcr_table[input_1].isna(), input_1] = tcr_table[Calculated_1]
+        tcr_table[input_1] = tcr_table[input_1].str.upper()
+        Calculated_2 = input_2.replace("Curated", "Calculated")
+        tcr_table.loc[tcr_table[input_2].isna(), input_2] = tcr_table[Calculated_2]
+        tcr_table[input_2] = tcr_table[input_2].str.upper()
+        return tcr_table
+
+    tcr_table = replace_curated("Chain 1 CDR3 Curated", "Chain 2 CDR3 Curated")
+    tcr_table = replace_curated("Curated Chain 1 V Gene", "Curated Chain 2 V Gene")
+    tcr_table = replace_curated("Curated Chain 1 D Gene", "Curated Chain 2 D Gene")
+    tcr_table = replace_curated("Curated Chain 1 J Gene", "Curated Chain 2 J Gene")
```

Comment:
Good idea to make a function! 

One point here is that you modify the object `tcr_table` in the function directly.
This is perfectly fine in this case, but you don't need to return and reassign the table. 

codestyle nitpick: `Calculated_` --> `calculated_` as it represents a variable, not a class.

Reply:
Thank you for your suggestions, but I am not sure if I understand correctly. If I don't use return, the function doesn't return anything, if I don't assign the output to a variable, it is not stored. What should I change?

Okay I will change the variable name, also, I need to change "tcr_table" to "iedb_df", "tcr_table" doesn't exist anymore since you changed the variable name earlier.

Reply:
There's an important difference between primitive types (int, etc.) and objects (e.g. dict, DataFrame, ...).
An object is always passed as a reference, that means you can change the content of the original object. 

Check out the following example: 

```python
a = 5
b = {"b": 5}

def do_stuff():
    a = 7
    b["b"] = 7
    
do_stuff()
print(a)
print(b)
```

Maybe also check out [this section](https://fabienmaussion.info/scientific_programming/week_03/01-Import-Scopes.html) of the python course at uni innsbruck. Maybe you have even a chance taking it, the online version looks excellent.

Reply:
Okay, this example is very clear, I understand. I updated the code. Thanks for the Link, the course seems indeed well-made.